flumes/discoverer.py

```diff
@@ -291,8 +292,19 @@ def on_discovered(self, discoverer, info, error):
             self.discovery_done()
             return
 
-        if error:
+        if error and type(error) == gi.repository.GLib.GError:
```

Comment:
```suggestion
        if type(error) == gi.repository.GLib.GError:
```
Since you have added a more specific condition you can get rid of the previous one that was simply checking the existence of error. I mean that if error is empty the new condition will fail anyway